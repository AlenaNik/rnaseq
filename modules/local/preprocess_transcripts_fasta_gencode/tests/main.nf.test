nextflow_process {

    name "Test Process PREPROCESS_TRANSCRIPTS_FASTA_GENCODE"
    script "../main.nf"
    process "PREPROCESS_TRANSCRIPTS_FASTA_GENCODE"

    test("sarscov2 fasta") {

        when {
            process {
                """
                input[0] = file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
