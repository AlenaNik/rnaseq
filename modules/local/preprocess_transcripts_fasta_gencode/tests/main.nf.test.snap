{
    "sarscov2 fasta": {
        "content": [
            {
                "0": [
                    "genome.fixed.fa:md5,6e9fe4042a72f2345f644f239272b7e6"
                ],
                "1": [
                    "versions.yml:md5,e2c7f636c472887a914b128d1444737e"
                ],
                "fasta": [
                    "genome.fixed.fa:md5,6e9fe4042a72f2345f644f239272b7e6"
                ],
                "versions": [
                    "versions.yml:md5,e2c7f636c472887a914b128d1444737e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-06T16:11:04.533974"
    }
}